src/event_pusher/mod_event_pusher_hook_translator.erl

```diff
@@ -22,36 +22,33 @@
 
 -export([add_hooks/1, delete_hooks/1]).
 
--export([user_send_packet/4,
-         filter_local_packet/1,
-         user_present/2,
-         user_not_present/5,
-         unacknowledged_message/2]).
-
--ignore_xref([filter_local_packet/1, unacknowledged_message/2, user_not_present/5,
-              user_present/2, user_send_packet/4]).
+-export([user_send_packet/3,
+         filter_local_packet/3,
+         user_present/3,
+         user_not_present/3,
+         unacknowledged_message/3]).
 
 %%--------------------------------------------------------------------
 %% gen_mod API
 %%--------------------------------------------------------------------
 
 -spec add_hooks(mongooseim:host_type()) -> ok.
 add_hooks(HostType) ->
-    ejabberd_hooks:add(hooks(HostType)).
+    gen_hook:add_handlers(hooks(HostType)).
 
 -spec delete_hooks(mongooseim:host_type()) -> ok.
 delete_hooks(HostType) ->
-    ejabberd_hooks:delete(hooks(HostType)).
+    gen_hook:delete_handlers(hooks(HostType)).
 
 %%--------------------------------------------------------------------
 %% Hook callbacks
 %%--------------------------------------------------------------------
 -type routing_data() :: {jid:jid(), jid:jid(), mongoose_acc:t(), exml:element()}.
--spec filter_local_packet(drop) -> drop;
-                         (routing_data()) -> routing_data().
-filter_local_packet(drop) ->
-    drop;
-filter_local_packet({From, To, Acc0, Packet}) ->
+-spec filter_local_packet(drop, _, _) -> {ok, drop};
+                         (routing_data(), _, _) -> {ok, routing_data()}.
+filter_local_packet(drop, _, _) ->
+    {ok, drop};
```

Comment:
For now, it's fine to leave it like this, but please collect the list of all the hook handlers that use the trick like this (special Acc value to stop processing). gen_hook allows handler to stop hook processing in a nicer way. We should get rid of this practice once the full conversion is done.